### PR TITLE
The relative_path field of PULP_MANIFEST can now contain commas

### PR DESCRIPTION
**Pull request by MichalPysik**
Commas are a valid filename characters in both Linux and Windows
file systems, but the previous way of generating manifest file
did not	allow them. This commit	fixes this issue, and introduces
a test for this	very specific edge-case. The test is split into
all three possible sub-cases: comma in filename, comma in folder
name, comma(s) in both.

closes pulp#630

### CODE REVIEW

**Comment by lubosmj** [comment]
@MichalPysik, to get rid of the error, please, replace "closes pulp#630" with "closes #630" in the commit message.

**Comment by MichalPysik** [comment]
> @MichalPysik, to get rid of the error, please, replace `closes pulp#630` with `closes #630` in the commit message.

@lubosmj thanks a lot!

**Comment by MichalPysik** [comment]
> 
That was my initial solution, but we discussed further and decided that since each part should also have a separate check (filepath is relative and refers to a file, not folder, therefore cannot start or end with '/', I'm also not sure which characters can and cannot be part of a digest), it might be more elegant to perform the check in a single line.

**Comment by dralley** [comment]
On second thought I doubt we actually want that kind of granular feedback.  I still have mixed feelings about the regex but I can live with it if we feel like it's a more complete approach.  If we do go with it, let's make sure it's well documented what each portion is doing so that breaking out a regex reference isn't necessary :)

 I'll let others weigh in.

**Comment by bmbouter** [comment]
I'm not sure what we loose by not using the regex, but I'm hesitant to use the regex.

**Comment by MichalPysik** [comment]
@bmbouter @dralley The regex made sense when I wasn't sure whether the digest can also contain commas, but since now I know it should be in the form of a hexadecimal number, I will rewrite the code tommorow using rsplit() and an individual check for each part, which should also make it more readable.

File: pulp_file/manifest.py [hunk 1]
```diff
@@ -49,8 +51,12 @@ def parse(line):
             ValueError: on parsing error.
 
         """
-        part = [s.strip() for s in line.content.split(",")]
-        if len(part) != 3:
+        relative_path, digest, size = [s.strip() for s in line.content.rsplit(",", maxsplit=2)]
```
Comment (mdellweg) [inline pulp_file/manifest.py:54]: The subtle problem with this solution is, if the line does not contain enough commas, it will fail here and not raise the ValueError.

BTW: This would have been covered nicely by the original full-line regex.

File: pulp_file/manifest.py [hunk 2]
```diff
@@ -49,8 +51,12 @@ def parse(line):
...
+        if (
+            not fullmatch(r"^[^/]+(/[^/]+)*$", relative_path)
```
Comment (mdellweg) [inline pulp_file/manifest.py:56]: Can you explain, why you use this regex? IMHO it's still too complicated.

File: pulp_file/manifest.py (hunk 1 above)
Comment (MichalPysik) [reply pulp_file/manifest.py:54]: Thanks for the heads up, I didn't notice that, I will modify that.

File: pulp_file/manifest.py (hunk 2 above)
Comment (MichalPysik) [reply pulp_file/manifest.py:56]: Because apart from checking that both the first and the last character is different from '/', I also wanted to check that there is no "//", for example "aaa///ff" should be invalid.

Comment (mdellweg) [reply pulp_file/manifest.py]: ahh, i see.

**Review by mdellweg — APPROVED** [review]

**Review by dralley — APPROVED** [review]
